dataland-api-key-manager/src/main/kotlin/org/dataland/datalandapikeymanager/services/ApiKeyManager.kt

```diff
@@ -0,0 +1,231 @@
+            val activityStatus =
+                (expiryDateOfApiKey ?: currentTime) >= currentTime
+            logger.info(
+                "Validated Api Key with encoded salt ${storedHashedAndBase64EncodedApiKeyOfUser.saltBase64Encoded} " +
+                    "and calculated hashed and encoded value $receivedApiKeyHashedAndBase64Encoded. " +
+                    "The activity status of the API key is $activityStatus."
+            )
+            var validationMessageToReturn = validationMessageSuccess
+            if (!activityStatus) {
+                validationMessageToReturn = validationMessageExpiredApiKey
+            }
+            return storedHashedAndBase64EncodedApiKeyOfUser.apiKeyMetaInfo.copy(
+                active = activityStatus,
+                validationMessage = validationMessageToReturn
+            )
+        }
+    }
+
+    /**
+     * Validates a specified api key
+     * @param receivedApiKey the received API key to be validated
+     * @return the found api keys meta info
+     */
+    fun validateApiKey(receivedApiKey: String): ApiKeyMetaInfo {
+        val receivedAndParsedApiKey = apiKeyPrevalidator.parseApiKey(receivedApiKey)
+
+        val storedHashedAndBase64EncodedApiKeyOfUserOptional = storedHashedAndBase64EncodedApiKeyRepository.findById(
+            receivedAndParsedApiKey.parsedKeycloakUserIdBase64Encoded
+        )
+
+        if (storedHashedAndBase64EncodedApiKeyOfUserOptional.isEmpty) {
+            logger.info(
+                "Dataland user with the encoded Keycloak user Id " +
+                    "${receivedAndParsedApiKey.parsedKeycloakUserIdBase64Encoded} has no API key registered."
+            )
+            return ApiKeyMetaInfo(active = false, validationMessage = validationMessageNoApiKeyRegistered)
+        }
+
+        val storedHashedAndBase64EncodedApiKeyOfUser =
+            storedHashedAndBase64EncodedApiKeyOfUserOptional.get().toApiModel()
+        val salt = EncodingUtils.decodeFromBase64(storedHashedAndBase64EncodedApiKeyOfUser.saltBase64Encoded)
+        val receivedApiKeyHashedAndBase64Encoded = encodeToBase64(hashString(receivedApiKey, salt))
+
+        return checkIfApiKeyForUserIsCorrectAndReturnApiKeyMetaInfoWithActivityStatus(
+            receivedApiKeyHashedAndBase64Encoded,
+            storedHashedAndBase64EncodedApiKeyOfUser,
+            receivedAndParsedApiKey.parsedKeycloakUserIdBase64Encoded
+        )
+    }
+
+    /**
+     * Revokes the API key of the authenticating user
+     * @return the result of the attempted revocation as a status flag and a message
+     */
+    fun revokeApiKey(): RevokeApiKeyResponse {
+        val authentication = getAuthentication()
+        val keycloakUserIdBase64Encoded = getKeycloakUserIdBase64Encoded(authentication)
+        val revokementProcessSuccessful: Boolean
+        val revokementProcessMessage: String
+
+        if (storedHashedAndBase64EncodedApiKeyRepository.findById(
```

Comment:
Für "Suche den API Key eines Users und logge einen Fehler wenn er nicht da ist" könnte man eine Funktion einführen (es wird zumindest 2 mal benutzt)

Reply:
Ich schau es mir jetzt an.

Reply:
Meines Erachtens haben die zwei Stellen nur das checken auf "isEmpty()" gemeinsam.
Die darauffolgende Aktion jedoch ist sehr unterschiedlich und lässt sich nicht vereinen.
D.h. man könnte nur den Check in eine zentral Funktion stecken.
Aber selbst das würde ich lieber nicht machen, weil man bei dem ersten Check das Optional vorher schon gespeichert hat und in den Check nur noch reinstopft, und beim zweiten Check ruft man das Optional nur für den Check ab.

Aber lass uns gerne darüber reden wenn ich falsch liege. Vllt übersehe ich da etwas.

Reply:
In der aktuellen Form ergibt der Kommentar keinen Sinn mehr